src/FsCheck/Runner.fs

```diff
@@ -150,45 +158,45 @@ module Runner =
             else xs.Add <| EndShrink result; xs :> seq<_> |> OutcomeSeqOrFuture.Value
         iter result shrinks
 
-    let private test initSize resize rnd0 gen =
-        let rec test' initSize resize rnd0 ((Gen eval) as gen) =
-            seq { 
-                let rnd1,rnd2 = Random.split rnd0
-                let newSize = resize initSize
-                //result forced here!
-                let result, shrinks =
-                    try
-                        let (MkRose (Lazy result, shrinks)) = (eval (newSize |> round |> int) rnd2).Value
-                        result, shrinks
-                        //printfn "After generate"
-                        //problem: since result.Ok is no longer lazy, we only get the generated args _after_ the test is run
-                    with :? DiscardException -> 
-                        Res.rejectedV, Seq.empty
-
-                yield Generated result.Arguments
-
-                match result.Outcome with
-                    | Outcome.Rejected -> 
-                        yield Failed result 
-                    | Outcome.True -> 
-                        yield Passed result
-                    | o when o.Shrink -> 
-                        yield Falsified result
-                        yield! shrinkResultValue result (shrinks.GetEnumerator ())
-                    | _ ->
-                        yield Falsified result
-                        yield EndShrink result
-                yield! test' newSize resize rnd1 gen
-            }
+    let stepsSeq resize =
+        Seq.unfold (fun (initSize, rnd) ->
+            let rnd1, rnd2 = Random.split rnd
+            let newSize = resize initSize
+            Some ((rnd1, rnd2, newSize), (newSize, rnd1)))
+
+    let private test initSize resize rnd0 gen =    
+        let test' ((Gen eval) as gen) (rnd1, rnd2, newSize) =
```

Comment:
I was pursuing goal to keep this function applicable to regular test run as well as replay. That led to ugly "two-cell memory" with `preLastSeed` and `preLastSize` in `runner`. Looking at what being done I don't know whether that was good idea. We need those `preLastXXX` because initial size and seed are not used to produce args, only for seeding in `stepsSeq`, so we have to report rnd and size previous to those led to fault which will be seeded in `stepsSeq`. So maybe exposing inner `test'` for replaing purposes is viable alternative.